common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -281,14 +285,26 @@ def _gated_content_staff_banner(self):
             if content_milestones and self.runtime.user_is_staff:
                 return banner_text
 
+    def _get_course_end(self):
+        """
+        Helper to get course end, because I can't figure out an easier way to do it. Open to suggestions.
+        """
+        current = self
+        while current.location.block_type != 'course':
+            current = current.get_parent()
+            if current == None:
+                raise Exception("Key assumption violated - went past root of course tree without hitting course block")
+        return current.end
+
     def _can_user_view_content(self):
         """
         Returns whether the runtime user can view the content
         of this sequential.
         """
+        hidden_date = self._get_course_end() if getattr(self, 'self_paced', False) else self.due
```

Comment:
An alternative to creating tests with a real runtime to support inheritable fields, I believe you can just change the above line to:

```
course = self._get_course()
hidden_date = course.end if course.self_paced else self.due
``` 

That is, get the `self_paced` value off of the course - rather than off of the sequence module.  We have verified with Shelby that the self_paced value isn't intended to be inherited or overridden at the block level.